### PR TITLE
Manual merge pr 63

### PR DESCRIPTION
**Pull request by setmar**
Added support for setting static dt, and unit tests. Also done some minor refactoring for adoption to new code structure.

### CODE REVIEW

**Comment by babrodtk** [comment]
This is not a PR on the variable_dt branch

**Comment by babrodtk** [comment]
I cannot merge this, it conflicts. Please fix the conflicts, and I will take a look at it

**Comment by setmar** [comment]
Please make PR with branch variable_dt in metno/gpu-ocean as base after merging.

File: python/SWESimulators/KP07.py [hunk 1]
```diff
@@ -41,7 +41,10 @@ def __init__(self, \
                  dx, dy, \
                  g, f=0.0, r=0.0, \
                  t=0.0, \
-                 theta=1.3, use_rk2=True, coriolis_beta=0.0, \
+                 dt=None, \
+                 dt_scale=1.0, \
```
Comment (babrodtk) [inline python/SWESimulators/KP07.py:45]: Why does this appear twice in the parameters list?

**Comment by babrodtk** [comment]
This should be rebased, creating a new PR

File: python/SWESimulators/KP07.py (hunk 1 above)
Comment (setmar) [reply python/SWESimulators/KP07.py:45]: That would be a mistake.

**Pull request closed without merging**